blockchain/Validation.go

```diff
@@ -2169,6 +2174,65 @@ func AcceptToMemoryPool(param *msg.BitcoinParams, pool *mempool.Mempool, state *
 	return true
 }
 
+func GetTransaction(param *msg.BitcoinParams, txid *utils.Hash, txOut *model.Tx, hashBlock *utils.Hash, fAllowSlow bool) (ret bool) {
+	var pindexSlow *model.BlockIndex
+	//todo:LOCK(cs_main)
+
+	ptx := mempool.GetTxFromMemPool(*txid)
+	if ptx != nil {
+		txOut = ptx
+		return true
+	}
+
+	if GfTxIndex {
+		var postx model.DiskTxPos
+		if Gpblocktree.ReadTxIndex(txid, &postx) {
+			file := OpenBlockFile(postx.BlockIn, true)
+			if file == nil {
+				return logger.ErrorLog("GetTransaction:OpenBlockFile failed")
+			}
+			ret = true
+			if err := recover(); err != nil {
+				logger.ErrorLog(fmt.Sprintf("%s: Deserialize or I/O error -%s", logger.TraceLog(), err))
+				ret = false
+			}
+			var header model.BlockHeader
+			header.Serialize(file)
+			file.Seek(0, 1)
```

Comment:
call file.Seek() parament is error; here should seek specific lenth in current file descriptor location。